ops/model.py

```diff
@@ -42,7 +43,7 @@ def get_relation(self, relation_name, relation_id=None):
                     return relation
             else:
                 # The relation may be dead, but it is not forgotten.
-                return Relation(relation_name, relation_id, self.unit, self._backend, self._cache)
+                return Relation(relation_name, relation_id, self._meta.relations[relation_name].role, self.unit, self._backend, self._cache)
```

Comment:
This does feel a bit strange. How is self.relations None, but we still have meta information about it?

This also *looks* like role is the odd-attribute-out. But it does feel like a property of a Relation to me.

Reply:
`self.relations` may be present but the relation may be gone (i.e. no relation with such a relation id per `relation-ids`).

The weirdness here is also in the [for/else construct](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops). 

`else` clause can be reached by using `get_relation` with a specific `relation_id` passed in while this relation id is gone at the time of a hook execution. So, a relation can be dead, but there will still be static metadata about it.